content/concepts/connections.md

```diff
@@ -0,0 +1,155 @@
+- **Initiator**: a peer that initiates a connection through a dial request.
+- **Responder**: a peer that listens to dial requests.
+- **Transport**: a mechanism of communication that creates a *Connection*, 
+  e.g., TCP, QUIC, WebRTC, etc.
+- **Stream**: an abstraction on top of a *Connection* representing a bidirectional 
+  communication channel between two *Peers*.
+- **Multiplexer**: a mechanism that manages multiple streams and ensures parallel 
+  messages reach the correct *Stream*, based on stream identification.
+- **Secure channel**: establishes a secure, encrypted, and authenticated channel 
+  over a *Connection*.
+- **Upgrader**: a mechanism that takes a *Connection* returned by a *Transport*, 
+  and performs protocol negotiation to set up a secure, multiplexed channel on 
+  top of which *Streams* can be opened.
+
+## Opening a libp2p connection
+
+A libp2p connection only requires two core operations: dialing and listening. 
+Both operations exist as interfaces in libp2p, where a transport protocol is used 
+to expose the dialing and listening interfaces. The listening interface allows a 
+peer to listen to connection requests from others, while the dialing interface 
+allows a peer to send connection requests to a listening peer. Most of the connection 
+initation process is abstracted through libp2p's interfaces.
+
+### How do peers prevent multiple connections from being established? 
+
+An interface called a [switch](/concepts/stream-multiplexing/#switch/swarm) 
+(and sometimes swarm) implements a basic dial queue that manages an application's 
+dialing and listening state. This prevents multiple, 
+simultaneous dial requests to the same peer. 
+
+### How do peers know what transports other peers support?
+
+Each libp2p node has information about the transports it supports.
+Peers can obtain this information by dialing a peer and listen 
+for updates on the network. More information about peers is available 
+on the [peers guide](/concepts/peers).
+
+### How do peers establish a connection once they find others to connect to?
+
+There are different approaches that peers can take to establish a connection.
+The dialing peer, known as the *Initiator* of the connection, sends a dialing request 
+to the listening peer, known as the *Responder* of the connection, via the Initiator's 
+multiaddr over the transport. To accept connections, a libp2p application registers 
+handler functions for protocols using their protocol ID.
+
+More information on multiaddrs, peers, and protocols is available on the
+[addressing](/concepts/addressing), [peers](/concepts/peers),
+[protocol](/concepts/protocols) guides, respectively.
+
+### How does the switch decide on which transport protocol to use?
+
+The switch conducts a protocol negotiation process between two peers through an 
+interface known as [`multistream-select`](https://github.com/multiformats/multistream-select).
+In particular, the multicodec is negotiated between two peers.
```

Comment:
I don't think this is accurate.

A node learns about a peers multiaddresses. Based on these multiaddresses the local node dials one or more addresses, i.e. decides on one or more transport protocols.